Remove global_step argument

tagomatech opened the pull request:
Following discussion in issue [#113](https://github.com/tensorflow/agents/issues/113), replace  `self._optimizer.apply_gradients(grads_and_vars, global_step=self.train_step_counter)` with     `self._optimizer.apply_gradients(grads_and_vars)` to ensure TF2 compatibility.

tagomatech committed:
Remove global_step argument

Replace  `self._optimizer.apply_gradients(grads_and_vars, global_step=self.train_step_counter)` with     `self._optimizer.apply_gradients(grads_and_vars)` to ensure TF2 compatibility.

tagomatech committed:
Merge pull request #1 from tagomatech/tagomatech-patch

Remove global_step argument from call to apply_gradients method

sguada commented on tf_agents/agents/dqn/dqn_agent.py:
```diff
@@ -294,8 +294,7 @@ def _train(self, experience, weights):
-    self._optimizer.apply_gradients(grads_and_vars,
-                                    global_step=self.train_step_counter)
+    self._optimizer.apply_gradients(grads_and_vars)
 
```
add

```
# Increment train_step_counter
self.train_step_counter.assign_add(1)
```

tagomatech replied:
You are perfectly right @sguada. Thank you. Done.

tagomatech committed:
Add self.train_step_counter.assign_add(1

tagomatech committed:
Merge pull request #2 from tagomatech/tagomatech-patch-1

Add self.train_step_counter.assign_add(1)

sguada commented:
Sorry there was a change in this file, so you need to sync to head and merge the changes.

ebrevdo commented:
I thought we have a PR that selectively passes global_step iff the optimizer is a v1 optimizer.

tagomatech committed:
Update dqn_agent.py

tagomatech commented:
> Sorry there was a change in this file, so you need to sync to head and merge the changes.

The file has changed substantially, indeed. I made changes requested. Hopefully I did it the right way. Thank you for all the work and patience.

tagomatech committed:
Merge branch 'master' into master

sguada commented on tf_agents/agents/dqn/dqn_agent.py:
```diff
@@ -14,9 +14,7 @@
 # limitations under the License.
 
 """A DQN Agent.
-
```
Undo all the space removal.

tagomatech replied:
You were right, several spaces disappeared. Thank you.

sguada commented on tf_agents/agents/dqn/dqn_agent.py:
```diff
@@ -364,8 +349,10 @@ def _train(self, experience, weights):
       eager_utils.add_gradients_summaries(grads_and_vars,
                                           self.train_step_counter)
     training.apply_gradients(self._optimizer,
-                             grads_and_vars,
```
replace all this with
```

self._optimizer.apply_gradients(grads_and_vars)

# Increment train_step_counter
self.train_step_counter.assign_add(1)
```

tagomatech replied:
Apologies for the delay. I just hadn't seen your new replacement request. Thank you.

sguada reviewed (changes requested):
Undo all the space removal in the file.

tagomatech committed:
Merge pull request #3 from tensorflow/master

Also summarize non-trainable weights in DQN.

tagomatech committed:
Update dqn_agent.py

tagomatech committed:
Update dqn_agent.py

tagomatech committed:
Merge pull request #4 from tensorflow/master

update from tensorflow/agents

tagomatech committed:
Update dqn_agent.py

Following @sguada's [comment](https://github.com/tensorflow/agents/pull/231#discussion_r338835204)

tagomatech committed:
Update dqn_agent.py

tagomatech committed:
port train_eval.py to TF2.0

tagomatech committed:
ported train_eval_test.py to tf2.0

tagomatech committed:
Create __init__.py

tagomatech committed:
Update train_eval.py

tagomatech committed:
Update train_eval.py

tagomatech committed:
Update __init__.py

tagomatech committed:
Merge pull request #7 from tagomatech/tagomatech-patch-4

Update __init__.py

tagomatech committed:
Merge pull request #6 from tagomatech/tagomatech-patch-3

Update train_eval.py

tagomatech committed:
Merge pull request #5 from tagomatech/tagomatech-patch-2

Update train_eval.py

sguada reviewed (changes requested):
Can you revert all the changes you made un-related to the the global_step change?

tagomatech committed:
Update Periodically.md

tagomatech committed:
remove example dqn v3

tfboyd commented:
I feel bad this is the 3rd PR closed you authored.  You did a bunch of commits and worked to address the issues.  We have gotten much better the past 3 months at accepting PRs.  This one looks to be stale as the underlying code to apply the gradients looks to be using a utility class now.  Your solution was correct, we failed to process it fast enough.  Super sorry and we are getting better.

tagomatech commented:
no worries. thanks! and thanks for TF agents.
